design/ironic_authentication.md

```diff
@@ -0,0 +1,329 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Ironic authentication
+
+## Status
+
+provisional
+
+## Summary
+
+Currently, Ironic and Ironic inspector APIs are unauthenticated to allow BMO to
+interact directly and without authentication with Ironic. However this opens a
+very wide attack target as anything running on a node that has an IP address on
+the provisioning network is able to access both APIs and hence control Ironic.
+This proposal modifies the ironic setup and BMO implementation to set up an
+authentication mechanism. This proposal covers a simple username/password
+authentication, a mutual TLS authentication setup and a keystone based
+authentication. In addition, it enables HTTPS for most of the APIs.
+
+Some calls between ironic and ironic inspector or between IPA and ironic or
```

Comment:
How mTLS would apply for the communication between APIs and conductor? I'm wondering how you will handle things in the jsonrpc

Reply:
yes, all communications between the different ironic components go through the localhost. We'll need to make sure that nothing else is exposed, other than the API

Reply:
actually, I am now considering exposing the json rpc interface behind mTLS too. That would allow us to break the ironic pod into multiple smaller pieces, and have conductor separated from API, as a first step towards scaling Ironic.